modules/common/Riot.py

```diff
@@ -49,32 +54,17 @@ def run_riot(self, owl_file, dir_output, json_file, owl_jq):
         :param owl_jq: JQ filtering for the JSON-LD conversion of the OWL file
         :return: destination file path of the conversion + filtering for the given OWL file
         """
-        path_output = os.path.join(dir_output, json_file)
-        # Set JVM memory limits
-        run_env = os.environ.copy()
-        run_env["_JAVA_OPTIONS"] = "-Xms4096m -Xmx8192m"
+        path_output = Path(dir_output).joinpath(json_file)
+        riot_outfile = Path(dir_output).joinpath(riot_out)
+        riot_cmd = f"{self.riot_cmd} --output JSON-LD {owl_file} > {riot_outfile}"
+        jq_cmd = f"{self.jq_cmd} -r '{owl_jq}' {riot_outfile} > {path_output}"
         try:
-            with open(path_output, "wb") as json_output, \
-                    subprocess.Popen([self.riot_cmd, "--output", "JSON-LD", owl_file], env=run_env,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as riot_process, \
-                    subprocess.Popen([self.jq_cmd, "-r", owl_jq], env=run_env,
-                                     stdin=riot_process.stdout,
-                                     stdout=subprocess.PIPE,
-                                     stderr=subprocess.PIPE) as jq_process:
-                json_output.write(jq_process.stdout.read())
-                logger.error(f"RIOT stderr -> '{riot_process.stderr.read()}'")
-                logger.error(f"JQ stderr -> '{jq_process.stderr.read()}'")
-        except EnvironmentError as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
-            logger.error(msg_err)
-            raise RiotException(msg_err)
-        except Exception as e:
-            msg_err = "When running RIOT for OWL file '{}', "
-            "with destination path '{}' and JQ filter '{}', "
-            "the following error occurred: '{}'".format(owl_file, path_output, owl_jq, e)
+            subproc(cmd=riot_cmd, env=self.get_running_environment())
+            logger.debug("riot command completed.")
+            subproc(cmd=jq_cmd)
+            logger.debug("jq command completed.")
+        except CustomSubProcException as err:
+            msg_err = f"The following error occurred: '{err}'"
```

Comment:
Taking into account my comments on the changes to _Utils.py_ file, I think we could just pass the raised error condition, to whatever code called *run_riot()*
Another approach would be to further customize the error message received through the exception into something like
```
f"FAILED to run RIOT on file '{owl_file}' and JQ with filter '{owl_jq}' due to the following error: '{err}'"
```

This way, the log message contains all the information we need to know what happened, with the details that belong to this level of abstraction.